pkg/client/client.go

```diff
@@ -115,99 +187,64 @@ func (n *NutanixClientHelper) GetClientFromEnvironment(ctx context.Context, nuta
 		}
 		npe.AdditionalTrustBundle.Namespace = capxNamespace
 	}
-	providers = append(providers, kubernetesEnv.NewProvider(
-		*npe,
-		n.secretInformer,
-		n.configMapInformer))
-	// init env with providers
-	env := environment.NewEnvironment(
-		providers...,
-	)
-	// fetch endpoint details
-	me, err := env.GetManagementEndpoint(envTypes.Topology{})
+
+	return kubernetesEnv.NewProvider(*npe, n.secretInformer, n.configMapInformer), nil
+}
+
+func Build(creds prismgoclient.Credentials, additionalTrustBundle string) (*nutanixClientV3.Client, error) {
+	cli, err := buildClientFromCredentials(creds, additionalTrustBundle)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get management endpoint object: %w", err)
+		return nil, err
 	}
-	creds := prismgoclient.Credentials{
-		URL:      me.Address.Host,
-		Endpoint: me.Address.Host,
-		Insecure: me.Insecure,
-		Username: me.ApiCredentials.Username,
-		Password: me.ApiCredentials.Password,
+	// Check if the client is working
+	_, err = cli.V3.GetCurrentLoggedInUser(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current logged in user with client: %w", err)
 	}
-
-	return n.GetClient(creds, me.AdditionalTrustBundle)
+	return cli, nil
 }
 
-func (n *NutanixClientHelper) GetClient(cred prismgoclient.Credentials, additionalTrustBundle string) (*nutanixClientV3.Client, error) {
-	if cred.Username == "" {
-		return nil, fmt.Errorf("could not create client because username was not set")
+func buildClientFromCredentials(creds prismgoclient.Credentials, additionalTrustBundle string) (*nutanixClientV3.Client, error) {
```

Comment:
Split this out to be unit testable, it would have been more difficult to get the `cli.V3.GetCurrentLoggedInUser(context.Background())` call working with some fake http server.